scripts/dataplan_package/llc_initial.py

```diff
@@ -0,0 +1,162 @@
+import time
+from typing import Any
+
+_LOGGER = logging.getLogger(__name__)
+
+_TEMPLATE = Path(__file__).resolve().parent / "template"
+_TEMPLATE_TESTS = Path(__file__).resolve().parent / "template_tests"
+_TEMPLATE_SAMPLES = Path(__file__).resolve().parent / "template_samples"
+
+
+def check_parameters(
+        output_folder: str,
+) -> None:
+    # check output_folder exists or not. If not, create it.
+    output = Path(output_folder)
+    if not os.path.exists(output):
+        _LOGGER.info(f'{output} does not exist and try to create it')
+        os.makedirs(output)
+        _LOGGER.info(f'{output} is created')
+
+
+def generate_test_sample(template_path: Path, target_path: Path, **kwargs: Any) -> None:
+    if not os.path.exists(target_path):
+        os.makedirs(target_path)
+    env = Environment(loader=FileSystemLoader(template_path), keep_trailing_newline=True)
+    for template_name in env.list_templates():
+        _LOGGER.info(f"generate file: {template_name}")
+        template = env.get_template(template_name)
+        result = template.render(**kwargs)
+        with open(target_path / template_name, "w") as fd:
+            fd.write(result)
+
+
+def generate_swagger_readme(work_path: str, env: Environment, **kwargs: Any) -> Path:
+    _LOGGER.info("Building swagger readme")
+    # check path exists
+    swagger_path = Path(work_path) / Path('swagger')
+    if not os.path.exists(swagger_path):
+        os.makedirs(swagger_path)
+
+    # render file
+    template = env.get_template('swagger_README.md')
+    result = template.render(**kwargs)
+    swagger_readme = swagger_path / Path('README.md')
+    with open(swagger_readme, 'w') as fd:
+        fd.write(result)
+    return swagger_readme
+
+
+def build_package(**kwargs) -> None:
+    # prepare template render parameters
+    output_folder = kwargs.get("output_folder")
+    package_name = kwargs.get("package_name")
+    swagger_readme_output = '../' + package_name.replace('-', '/')
+    kwargs['swagger_readme_output'] = swagger_readme_output
+    namespace = package_name.replace('-', '.')
+    kwargs['namespace'] = namespace
+    kwargs['date'] = time.strftime('%Y-%m-%d', time.localtime())
+    folder_list = package_name.split('-')
+    kwargs['folder_first'] = folder_list[0]
+    kwargs['folder_second'] = folder_list[1]
+    kwargs['folder_parent'] = Path(output_folder).parts[-2]
+    kwargs['test_prefix'] = folder_list[-1]
+
+    _LOGGER.info("Build start: %s", package_name)
+    check_parameters(output_folder)
+
+    # generate swagger readme
+    env = Environment(loader=FileSystemLoader(_TEMPLATE), keep_trailing_newline=True)
+    swagger_readme = generate_swagger_readme(output_folder, env, **kwargs)
+
+    # generate code with autorest and swagger readme
+    _LOGGER.info("generate SDK code with autorest")
+    check_call(f'autorest --version=3.4.5 --python --use=@autorest/python@5.8.4 --use=@autorest/modelerfour@4.19.2 --python-mode=update {swagger_readme}', shell=True)
```

Comment:
- I do not want to fix m4 in a cmdline, autorest should be responsible to do it
- we should us latest autorest.python
- we should make this tool generate with version-tolerant flags (talk to @iscai-msft )

Reply:
(1) To make package more stable, I think it's better to pin it like mgmt-plan package: https://github.com/Azure/azure-sdk-for-python/blob/6341dc07b2a81b16a403714ba34eaf17a2ae6ebb/swagger_to_sdk_config_autorest.json#L5.
(2) Of course, I will update the version before merge it.
(3) All the necessary configuration item is in swagger `README.md` so that customers needn't use complicated cmd:
![image](https://user-images.githubusercontent.com/70930885/141713937-5b313483-993f-4273-9af6-fe75ab952a03.png)